pkg/controller/container-runtime-config/container_runtime_config_controller.go

```diff
@@ -687,6 +739,28 @@ func (ctrl *Controller) syncImageConfig(key string) error {
 	} else if err != nil {
 		return err
 	}
+	// Find all ImageDigestMirrorSet objects
+	idmsRules, err := ctrl.idmsLister.List(labels.Everything())
+	if err != nil && errors.IsNotFound(err) {
+		idmsRules = []*apicfgv1.ImageDigestMirrorSet{}
+	} else if err != nil {
+		return err
+	}
+
+	// Find all ImageTagMirrorSet objects
+	itmsRules, err := ctrl.itmsLister.List(labels.Everything())
+	if err != nil && errors.IsNotFound(err) {
+		itmsRules = []*apicfgv1.ImageTagMirrorSet{}
+	} else if err != nil {
+		return err
+	}
+
+	if err := runtimeutils.RejectMultiUpdateMirrorSetObjs(icspRules, idmsRules, itmsRules); err != nil {
+		return err
+	}
+	for _, icsp := range icspRules {
```

Comment:
It’s a bit surprising that MCO is not calling `EditRegistriesConfig` with the raw ICSP data.

It _can_ work that way, of course, but MCO is the most frequent caller of that function, so having it call `EditRegistriesConfig` exactly the same way as the other callers would decrease the risk of MCO drifting away from the other mirror config consumers (like builds).

This code can still do its own conversion or merging for the purposes of `getValidBlockedAndAllowedRegistries` — maybe inside that function. Handling the ICSP/IDMS duality inside that function would also help us avoid needing the three copies of this loop.